x-pack/legacy/plugins/apm/public/context/MatchedRouteContext.tsx

```diff
@@ -4,11 +4,17 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 import React, { useMemo, ReactChild } from 'react';
-import { matchPath } from 'react-router-dom';
+import { matchPath, match } from 'react-router-dom';
 import { useLocation } from '../hooks/useLocation';
-import { BreadcrumbRoute } from '../components/app/Main/ProvideBreadcrumbs';
+import {
+  BreadcrumbRoute,
+  LocationMatch
+} from '../components/app/Main/ProvideBreadcrumbs';
 
-export const MatchedRouteContext = React.createContext<BreadcrumbRoute[]>([]);
+export const MatchedRouteContext = React.createContext<{
+  routes: BreadcrumbRoute[];
+  matchPath?: LocationMatch['match'];
```

Comment:
do we need to pass this through context? it's an import from `react-router-dom` no?

Reply:
I don't HAVE TO, but I need the matchPath to call the `breadcrumb` function. I decided to pass it here because I'm already looping through the routes and I have it in hands, otherwise I'd have to get it again from the route.